sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala

```diff
@@ -96,6 +96,9 @@ class ResolveCatalogs(val catalogManager: CatalogManager)
     case AlterNamespaceSetPropertiesStatement(NonSessionCatalog(catalog, nameParts), properties) =>
       AlterNamespaceSetProperties(catalog, nameParts, properties)
```

Comment:
Can you do `catalog.asNamespaceCatalog` so that it fails earlier if `catalog` doesn't support namespaces? The logical plan also needs to be updated to take a catalog of `SupportsNamespaces` instead of `CatalogPlugin`. Check `CreateNamespace` below. The reason why `DropNamespace` takes `CatalogPlugin` is that it needs to use both `TableCatalog` and `SupportsNamespaces`.

Reply:
good point!
updated, thanks.